src/taipy/gui/data/utils.py

```diff
@@ -12,16 +12,43 @@
 from __future__ import annotations
 
 import typing as t
+from abc import ABC, abstractmethod
 
 import numpy as np
 
-from ..utils.rdp import rdp_numpy
-
 if t.TYPE_CHECKING:
     import pandas as pd
 
 
-def _df_data_filter(dataframe: pd.DataFrame, x_column_name: t.Union[None, str], y_column_name: str, epsilon: int):
+class DecimatorABC(ABC):
+    def __init__(self, applied_threshold: t.Optional[int]) -> None:
+        self.applied_threshold = applied_threshold
+        super().__init__()
+
+    def _is_applicable(self, data: t.Any, nb_rows_max: int):
+        if self.applied_threshold is None:
+            if nb_rows_max < len(data):
+                return True
+        elif self.applied_threshold < len(data):
+            return True
+        return False
+
+    @abstractmethod
+    def decimate(self, data: np.ndarray) -> np.ndarray:
```

Comment:
We'll need a bit more explanation at this point.
That IS the real magic of all of this.

Reply:
Can you suggest anything to add

Reply:
First of all, I would rename the class 'Decimator', for short. The ABC aspect is an implementation detail.
And this class should definitively get some doc. At least a skeleton, full of TODO notes.

Then, this method really is the entry point of that feature. So we will need to document it thoughtfully. 
Again, TODO entries is fine to me.